File: SASMopubAdapterConfiguration.m
//
//  SASMopubAdapterConfiguration.m
//  Sample
//
//  Created by Julien Gomez on 15/01/2021.
//  Copyright © 2021 Smart AdServer. All rights reserved.
//

#import "SASMopubAdapterConfiguration.h"


@implementation SASMopubAdapterConfiguration

#pragma mark - Caching

+ (void)updateInitializationParameters:(NSDictionary *)parameters {
    
}

#pragma mark - MPAdapterConfiguration

- (NSString *)adapterVersion {
    return @"2.0";
}

- (NSString *)biddingToken {
    return nil;
}

- (NSString *)moPubNetworkName {
    return @"SmartAdServer";
}

- (NSString *)networkSdkVersion {
    return @"7.8";
}

- (void)initializeNetworkWithConfiguration:(NSDictionary<NSString *, id> *)configuration
                                  complete:(void(^)(NSError *))complete {

}

@end

File: SASMopubAdapterConfiguration.swift
//
//  SASMopubAdapterConfiguration.swift
//  SASMopubAdapterConfiguration
//
//  Created by Loïc GIRON DIT METAZ on 02/09/2021.
//

import Foundation
import SASDisplayKit
import MoPubSDK

/**
 Provides adapter information back to the SDK and is the main access point
 for all adapter-level configuration.
 */
@objc(SASMopubAdapterConfiguration)
class SASMopubAdapterConfiguration: MPBaseAdapterConfiguration {
    
    override var adapterVersion: String {
        return "3.0"
    }
    
    override var biddingToken: String? {
        return nil
    }
    
    override var moPubNetworkName: String {
        return "SmartAdServer"
    }
    
    override var networkSdkVersion: String {
        return "7.12"
    }
    
}
